src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java

```diff
@@ -34,4 +34,19 @@ public BookDto findById(Long id) {
                 .orElseThrow(() -> new EntityNotFoundException("Can't get book by id" + id));
         return bookMapper.toDto(book);
     }
+    
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(Long id, BookRequestDto bookDto) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Can`t find a book with id: " + id);
+        }
```

Comment:
delete method should validate the existence of book with provided `id` as well, that's why I said that we need a private method for that - to not duplicate the code